Modules/_functoolsmodule.c

```diff
@@ -560,37 +560,25 @@ functools_reduce(PyObject *self, PyObject *args)
         return NULL;
     }
 
-    if ((args = PyTuple_New(2)) == NULL)
-        goto Fail;
-
     for (;;) {
-        PyObject *op2;
-
-        if (args->ob_refcnt > 1) {
-            Py_DECREF(args);
-            if ((args = PyTuple_New(2)) == NULL)
-                goto Fail;
-        }
-
-        op2 = PyIter_Next(it);
+        PyObject *op2 = PyIter_Next(it);
         if (op2 == NULL) {
             if (PyErr_Occurred())
                 goto Fail;
             break;
         }
 
-        if (result == NULL)
+        if (result == NULL) {
             result = op2;
+        }
         else {
-            PyTuple_SetItem(args, 0, result);
-            PyTuple_SetItem(args, 1, op2);
-            if ((result = PyEval_CallObject(func, args)) == NULL)
+            PyObject *args[] = {result, op2};
+            if ((result = _PyObject_FastCall(func, args, 2)) == NULL) {
```

Comment:
Would you mind to move the assignment out of the if() please? "result = ...; if (result == NULL) ...".